src/App/App.tsx

```diff
@@ -2,40 +2,43 @@ import './App.scss'
 import React, { useState, useEffect } from 'react'
 import MainNav from '../MainNav/MainNav'
 import ParkContainer from '../ParkContainer/ParkContainer'
-// import { Route } from 'react-router-dom'
+import { Route, Switch, useLocation } from 'react-router-dom'
 import { nationalParks } from '../ParkData'
 import { parkRequest } from '../ParkContainer/npsApiCall'
 
-interface ChoosePark {
-  choosePark: (parkCode: string) => void
-}
-
-interface Park {
-  parkCode: string;
-}
-
 const App: React.FC = () => {
   const [parkCode, setParkCode] = useState('')
-  
+  const location = useLocation()
   const choosePark = (parkCodeId: string) => {
    setParkCode(parkCodeId)
   }
 
-  const generateRandomParkCode = () => {
-    let index = Math.floor(Math.random() * nationalParks.length)
-    setParkCode(nationalParks[index].parkCode)
-  }
-
   useEffect(() => {
-    if (parkCode === '') {
+    if (location.pathname === '/') {
       generateRandomParkCode()
+    } else {
+      setParkCode(location.pathname.split('/')[1])
     }
   }, [parkCode])
 
+  const generateRandomParkCode = () => {
+    let index = Math.floor(Math.random() * nationalParks.length)
+    setParkCode(nationalParks[index].parkCode)
+  }
+
   return (
     <main className="body">
       <MainNav choosePark={choosePark}/>
-      <ParkContainer parkCode={parkCode}/>
+      <Switch>
+        <Route
+          path={location.pathname}
+          render={ () => {
+            return (
+              <ParkContainer parkCode={parkCode}/>
+            )
+          }}
+        />
+      </Switch>
```

Comment:
Were we going to add a comment here to come back and add an Error component to this switch?